src/ktn_os.erl

```diff
@@ -0,0 +1,45 @@
+%% @doc Utility functions for
+-module(ktn_os).
+
+-export([command/1, command/2]).
+
+-type opts() :: #{log_fun => function()}.
+-type exit_status() :: integer().
+
+-spec command(iodata()) -> {exit_status(), string()}.
+command(Cmd) ->
+  Opts = #{log_fun => fun error_logger:info_msg/1},
+  command(Cmd, Opts).
+
+-spec command(iodata(), opts()) -> {exit_status(), string()}.
+command(Cmd, Opts) ->
+  PortOpts = [stream, exit_status, eof],
+  Port = open_port({spawn, shell_cmd()}, PortOpts),
+  erlang:port_command(Port, make_cmd(Cmd)),
+  get_data(Port, Opts, []).
+
+-spec get_data(port(), opts(), [string()]) -> {exit_status(), string()}.
+get_data(Port, Opts, Data) ->
+  receive
+    {Port, {data, NewData}} ->
+      case maps:get(log_fun, Opts, undefined) of
+        Fun when is_function(Fun) -> Fun(NewData);
+        undefined -> ok
+      end,
+      get_data(Port, Opts, [NewData | Data]);
+    {Port, eof} ->
+      port_close(Port),
+      receive
+        {Port, {exit_status, ExitStatus}} ->
+          {ExitStatus, lists:flatten(lists:reverse(Data))}
+      end
+  end.
```

Comment:
Can we have an `after` clause here? The value should be in `Opts` with a default of, say… an hour?